lucet-wasi/src/memory.rs

```diff
@@ -357,3 +357,80 @@ dec_enc_scalar!(
     enc_whence,
     enc_whence_byref
 );
+
+dec_enc_scalar!(
+    __wasi_subclockflags_t,
+    dec_subclockflags,
+    dec_subclockflags_byref,
+    enc_subclockflags,
+    enc_subclockflags_byref
+);
+
+dec_enc_scalar!(
+    __wasi_eventrwflags_t,
+    dec_eventrwflags,
+    dec_eventrwflags_byref,
+    enc_eventrwflags,
+    enc_eventrwflags_byref
+);
+
+dec_enc_scalar!(
+    __wasi_eventtype_t,
+    dec_eventtype,
+    dec_eventtype_byref,
+    enc_eventtype,
+    enc_eventtype_byref
+);
+
+dec_enc_scalar!(
+    __wasi_userdata_t,
+    dec_userdata,
+    dec_userdata_byref,
+    enc_userdata,
+    enc_userdata_byref
+);
+
+pub fn dec_subscription(
+    subscription: &wasm32::__wasi_subscription_t,
+) -> Result<host::__wasi_subscription_t, host::__wasi_errno_t> {
+    let userdata = dec_userdata(subscription.userdata);
+    let type_ = dec_eventtype(subscription.type_);
+    let u_orig = subscription.__bindgen_anon_1;
+    let u = match type_ {
+        wasm32::__WASI_EVENTTYPE_CLOCK => host::__wasi_subscription_t___wasi_subscription_u {
+            clock: unsafe {
+                host::__wasi_subscription_t___wasi_subscription_u___wasi_subscription_u_clock_t {
+                    identifier: u_orig.clock.identifier,
```

Comment:
still need a `dec_userdata` here

Reply:
My eyes can barely read anything any more, but I can't see any `userdata` here :/

Reply:
`identifier` is a `__wasi_userdata_t` :frowning_face:

Reply:
Done.